### DISCUSSION TITLE
BirdNET-Pi on Pi Zero-2W Works Great!

### DISCUSSION THREAD

**Discussion by dalogue1** [Nachtzuster/BirdNET-Pi#133] [General]
I have a fast BirdNET-Pi system but a friend wanted one set up but for much less cost.

I configured it just as recommended along with a homemade AOM 5024P microphone and a generic Sabrent USB Audio adapter.  As long as you load the lite version of the 64-bit Pi OS Bookworm for the Pi Zero 2W and with the recommended options it works great!  So do not hesitate if the Pi Zero 2W is good enough, because it is.

Note: I originally loaded the full OS and it was lousy because it was out of memory and terribly slow as could be expected.

**Comment by dalogue1** [comment]
BTW when you are installing the OS to the microSD card remember to preconfigure the SID and password for your WiFi, set your time zone, and very important enable ssh so you can login from your Mac/Linux using ssh or from iPad using Shelly, or Windows using PowerShell. The BirdNET-Pi interface also connects you to a file manager and database maintenance utility as well.

Of course, a faster Pi has many advantages but for many people like my friend they will not see any difference.  If you configure one for a friend,  it is very handy to use the ‘nmtui’ utility to easily configure multiple WiFi SIDs in advance. (Namely yours and your friend’s).  That way you can set up the BirdNET-Pi completely including broadcasting to BirdWeather and they can just plug it in with no configuration on their part at all. They can just plug it in to the power plug.

**Comment by Emmo213** [comment]
Nice, thanks for letting people know!

**Comment by MichaelPWalter** [comment]
Im having trouble setting up my RPi0W2. I increased the swap size as recommended but the entire rpi grinds to a halt after a couple of seconds. I can usually manage to connect to the birdnetpi.local. Is there a way to disable all but the core services ? I just need to be able to run Putty. And read out the gathered data at the end of the day. Ill slowly start to strip down all the services i dont need like wifi and hdmi.

**Reply by MichaelPWalter** [reply]
Seems like it could be caused by a subpar SD card. Ill try to find something faster and give it another try.

**Reply by MichaelPWalter** [reply]
Noticed swap size was still set to 200Mb on System Info panel. Reinstalled the OS and adjusted swap size manually now it works fine.

**Reply by dalogue1** [reply]
The installation instructions for the Pi-Zero 2W should have adjusted the Swap file size to 2048.  Check that the changes are all present.

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/4a0728cd-b5a0-4536-8568-51a87394252b)

**Reply by hannahilea** [reply]
Another way to double-check is to do `sudo nano /etc/dphys-swapfile` and make sure that the lines containing `CONF_MAXSWAP` are both uncommented (don't start with `#`!) and exactly `CONF_MAXSWAP=4096`, and `CONF_SWAPSIZE=2048`. (If they aren't, update them and resave the file, then do `sudo reboot`!)

The instructions for adjusting the swap size (as copied into the comment above this one) assume that the starting default is `CONF_SWAPSIZE=100`; if it isn't, this command won't update it AND won't warn or error (which makes sense---the command swaps out any instance of the first string with the second string, but doesn't confirm that the first string exists or that the value has been successfully updated). On my Pi-Zero 2W, the default was 150, so I didn't discover that I hadn't actually updated my swap size until I had to debug OOM issues later.

**Comment by dalogue1** [comment]
Absolutely.   I have the “performance” microSD from MicroCenter.

You should not have to disable anything at all.   Make sure you have loaded the 64 bit Lite version of the Pi OS   The system was horrible when I originally loaded the full version.

Follow the installation instructions for the Pi-Zero 2W and it will work fine.

https://github.com/mcguirepr89/BirdNET-Pi/wiki/RPi0W2-Installation-Guide

**Reply by MichaelPWalter** [reply]
I will give it a try. Do you think a 32-bit version works with BirdNET-Pi ?

**Reply by MichaelPWalter** [reply]
It doesnt

**Reply by dalogue1** [reply]
Nope

**Reply by dalogue1** [reply]
You may need to hunt a little for the Lite 64-bit version for the Pi-Zero

**Reply by dalogue1** [reply]
Actually it is just the "Raspberry Pi OS Lite (64-bit)" under the (other) choice.
![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/c68e2c6e-00fc-4219-8d4a-69f8d58f3e6f)
